cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -37,12 +37,31 @@ void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
       return cursum + _metadata->get_output_nesting_depth(chunk.src_col_schema);
     });
 
+  // Check to see if there are any string columns present. If so, then we need to get size info
+  // for each string page. This size info will be used to pre-allocate memory for the column,
+  // allowing the page decoder to write string data directly to the column buffer, rather than
+  // doing a gather operation later on.
+  // TODO: This step is somewhat redundant if size info has already been calculated (nested schema,
+  // chunked reader).
+  auto is_string_col = [](gpu::ColumnChunkDesc const& chunk) {
+    return (chunk.data_type & 7) == BYTE_ARRAY && (chunk.data_type >> 3) != 4 &&
+           chunk.converted_type != DECIMAL;
```

Comment:
I added the check for DECIMAL because I noticed [this](https://github.com/rapidsai/cudf/blob/5e12c25ca40c41f21fc80d4cd36713c514fabaa4/cpp/src/io/parquet/page_data.cu#L1085) and [this](https://github.com/rapidsai/cudf/blob/5e12c25ca40c41f21fc80d4cd36713c514fabaa4/cpp/src/io/parquet/page_data.cu#L2112). It appears that DECIMAL can be encoded as either BYTE_ARRAY or FIXED_LEN_BYTE_ARRAY. I only want to count up the BYTE_ARRAY columns that are UTF8.

Reply:
Oh, shame on me. It was me adding this feature and somehow I forgot binary can be read binary as decimals during the review.